### PR TITLE
Feature/2495/generate inventory

### PR DESCRIPTION
**Pull request by Richard-Hansen**
Partner PR: https://github.com/dockstore/dockstore-deploy/pull/194

Added a new lambda to run resource inventory generation. The original repo is in Python, so I followed through with Python. In hindsight it may have been better to convert this to typescript to match the other lambdas.

There is a lot of stuff being added with this PR, some special areas to note:
1. The original repo https://github.com/aws-samples/fedramp-integrated-inventory-workbook
2. The lambda source is under `fedramp-integrated-inventory-workbook/deployment/inventory`. mappers.py does the actual resource mapping from the AWS config JSONs.
3. There are a decent number of tests for the lambda as well run in CircleCI. The CircleCI `config.yml` also required a new command to package the Python source.
4. In `fedramp-integrated-inventory-workbook/tests/sample_config_query_results` there are example JSONs that represent what is provided by AWS config.

The current resources being mapped can be seen in `fedramp-integrated-inventory-workbook/deployment/inventory/readers.py`, which currently consists of `EC2` `ELB` `RDS` `Lambda` `VPC` and `S3`. We can add any other resource we want, as long as AWS config tracks it. I left out `Subnets` for the time being, as not all the information needed to map that resource is contained within the AWS config, so it would require linking multiple resources together to get the desired info.

### CODE REVIEW

File: fedramp-integrated-inventory-workbook/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
Comment (denis-yuen) [inline fedramp-integrated-inventory-workbook/.vscode/launch.json:2]: Weird, not legal json but I assume vscode is reading this properly?

File: fedramp-integrated-inventory-workbook/CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+## Code of Conduct
+This project has adopted the [Amazon Open Source Code of Conduct](https://aws.github.io/code-of-conduct).
```
Comment (denis-yuen) [inline fedramp-integrated-inventory-workbook/CODE_OF_CONDUCT.md:2]: No it hasn't :wink:

File: fedramp-integrated-inventory-workbook/LICENSE [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (denis-yuen) [inline fedramp-integrated-inventory-workbook/LICENSE:1]: We normally use Apache 2, think this is compatible for re-license but not a lawyer

File: fedramp-integrated-inventory-workbook/README.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
+# FedRAMP Integrated Inventory Workbook Generator
+
+## License
+
+This library is licensed under the MIT-0 License. See the LICENSE file.
```
Comment (denis-yuen) [inline fedramp-integrated-inventory-workbook/README.md:5]: So many licenses?

File: fedramp-integrated-inventory-workbook/deployment/inventory/reports.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
+# (c) 2019 Amazon Web Services, Inc. or its affiliates. All Rights Reserved.
```
Comment (denis-yuen) [inline fedramp-integrated-inventory-workbook/deployment/inventory/reports.py:1]: This too, hmm

File: fedramp-integrated-inventory-workbook/tests/sample_config_query_results/sample.json [hunk 6]
```diff
@@ -0,0 +1,294 @@
+[
```
Comment (denis-yuen) [inline fedramp-integrated-inventory-workbook/tests/sample_config_query_results/sample.json:1]: Think we can refer back to the original samples unless this is for testing

**Review by denis-yuen — COMMENTED** [review]
Some superficial observations, will need to swing back for deeper look

Think you could (and should) delete https://github.com/dockstore/dockstore-deploy/blob/develop/tools/inventory.sh and https://github.com/dockstore/dockstore-cli/blob/74eb5fc895cb5275cf54b70316fc26abc3c1e43d/support/src/main/java/io/dockstore/support/Inventory.java in favor of this off-the-shelf

File: fedramp-integrated-inventory-workbook/.vscode/launch.json (hunk 1 above)
Comment (coverbeck) [reply fedramp-integrated-inventory-workbook/.vscode/launch.json:2]: To add on, is VSCode even needed? Maybe delete this directory?

**Review by coverbeck — COMMENTED** [review]
There are several empty files. Are those needed/required?

> There are a decent number of tests for the lambda as well, although those are not currently being automatically run in CircleCI

Why not? Is it a lot of work to turn them on?

File: fedramp-integrated-inventory-workbook/deployment/inventory/reports.py (hunk 5 above)
Comment (Richard-Hansen) [reply fedramp-integrated-inventory-workbook/deployment/inventory/reports.py:1]: I kept a large portion of the original repo's 'extras' that aren't directly required for the lambda. If we don't think they're helpful as resources, and list a bunch of conflicting licenses, I can trim down as much as possible?

**Comment by Richard-Hansen** [comment]
> There are several empty files. Are those needed/required?
> 
> > There are a decent number of tests for the lambda as well, although those are not currently being automatically run in CircleCI
> 
> Why not? Is it a lot of work to turn them on?

The xls and __init__.py files are needed, I didn't realize the pngs are empty, I can remove those.

For the tests, I just haven't gotten there yet, but wanted to start getting feedback in case there are any big changes to the report that need to be made. I'll setup tests ASAP.

**Comment by david4096** [comment]
Moving this comment from deploy since it's more appropriate here: 

I took a look at the spreadsheet and this is great to see, especially since whether or not we set up end to end automation we have a way to generate a good starter. One nice-to-have would be to put the EC2 name under Function or Asset Tag.

Wondering how we maintain the "Record of Changes" page given this method.

File: fedramp-integrated-inventory-workbook/.gitignore [hunk 7]
```diff
@@ -0,0 +1,116 @@
+# Ignore MacOS folder
```
Comment (david4096) [inline fedramp-integrated-inventory-workbook/.gitignore:1]: this gitignore should probably just be merged into the one at the top of the repository unless there's a good reason to maintain a separate one?

File: fedramp-integrated-inventory-workbook/deployment/inventory/handler.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    return {'statusCode': 200,
+            'body': {
+                    'report': { 'url': report_url }
+                }
+            }
+
+if __name__ == "__main__":
+    class Context(object):
+        def __init__(self):
+            self.invoked_function_arn = "arn:aws-us-gov:lambda:us-east-1:123456789012:function:testing"
```
Comment (david4096) [inline fedramp-integrated-inventory-workbook/deployment/inventory/handler.py:22]: Any idea what this is here for? Looks like a mock

File: fedramp-integrated-inventory-workbook/deployment/inventory/mappers.py [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+                "software_product_name": f"{config_resource['configuration']['engine']}-{config_resource['configuration']['engineVersion']}",
+                "network_id": config_resource['configuration']['dBSubnetGroup']['vpcId'] if "dBSubnetGroup" in config_resource[
+                    'configuration'] else '',
+                "owner": _get_tag_value(config_resource["tags"], "owner"),
+                "location": config_resource["awsRegion"]}
+
+        return [InventoryData(**data)]
+
+
+class DynamoDbTableDataMapper(DataMapper):
```
Comment (david4096) [inline fedramp-integrated-inventory-workbook/deployment/inventory/mappers.py:182]: Doesn't hurt to leave in mappers we don't have but we don't have any plans to use Dynamo for anything.

**Comment by david4096** [comment]
Going to concur with the general idea of removing things that aren't necessary for us to operate/maintain this. If the tests are useful we should add them to CI. Great to see all this!

File: fedramp-integrated-inventory-workbook/.vscode/launch.json (hunk 1 above)
Comment (Richard-Hansen) [reply fedramp-integrated-inventory-workbook/.vscode/launch.json]: Removed lots of unused files, including some of the license ones and the .vscode directory.

File: fedramp-integrated-inventory-workbook/.gitignore (hunk 7 above)
Comment (Richard-Hansen) [reply fedramp-integrated-inventory-workbook/.gitignore]: Combined the two .gitignores, only took the required statements from the fedramp's .gitignore.

File: fedramp-integrated-inventory-workbook/deployment/inventory/handler.py (hunk 8 above)
Comment (Richard-Hansen) [reply fedramp-integrated-inventory-workbook/deployment/inventory/handler.py]: Initially thought it was used for testing, but upon further inspection it doesn't seem to do anything at the moment. Removing for now.

**Comment by Richard-Hansen** [comment]
CircleCI Python tests are now running.

**Review by coverbeck — APPROVED** [review]

**Review by david4096 — APPROVED** [review]

**Review by garyluu — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Review by GFJHogue — APPROVED** [review]

**Pull request merged**